### PR TITLE
Mbed trace

### PR DESCRIPTION
**Pull request by teetak01**
@yogpan01 @anttiylitokola @kuggenhoffen 

Update mbed Client to mbed-trace

### CODE REVIEW

File: module.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
   "homepage": "https://github.com/ARMmbed/mbed-client",
   "license": "Apache-2.0",
   "dependencies": {
-    "mbed-client-c": "^2.0.0"
+    "mbed-client-c": "^2.0.0",
+    "mbed-trace": "ARMmbed/mbed-trace"
```
Comment (yogpan01) [inline module.json]: Why is this added like this ? It should be added as yotta module and not as HEAD . Isn't that this module is now published ?

Comment (teetak01) [inline module.json]: The version number is currently 0.2.1 so I think I have to change that to something like:

Comment (yogpan01) [inline module.json]: Why still do it like mbed-trace": "ARMmbed/mbed-trace#0.2.1 and not like
mbed-trace": "0.2.1" ?

**Comment by anttiylitokola** [comment]
rebuild

**Comment by teetak01** [comment]
Rebased and cleaned commits.

**Comment by yogpan01** [comment]
LGTM

**Comment by yogpan01** [comment]
@anttiylitokola please review

**Comment by anttiylitokola** [comment]
Traces are missing when running smoketests. Does this need some changes to testapplication side to get client traces visible?

**Comment by teetak01** [comment]
yeap, all my pull requests to mbed Client depend on each other + related changes to mbed-client-testapp.

After this also mbed-client-linux-example and mbed-client-quickstart, etc. needs to be updated also.

I confirmed that with recent changes to mbed-trace there is no seg. fault anymore from mbed-client-linux-example if the mbed Client dependencies are updated without changes to mbed-client-linux-example.

**Comment by teetak01** [comment]
All smoketests PASS when locally linking the devel-branches to mbed-client-testapp#mbed-trace.

**Comment by teetak01** [comment]
Added also a PR for linux-example https://github.com/ARMmbed/mbed-client-linux-example/pull/36

mbed-client-examples, mbed-client-example-6lowpan or mbed-client-quickstart do not seem to directly use traces, so those examples should be unaffected.

**Comment by yogpan01** [comment]
rebuild
